### DISCUSSION TITLE
Issue Deploying HCIBox

### DISCUSSION THREAD

**Discussion by walezbee** [microsoft/azure_arc#1734] [Q&A] [answered]
I was trying to explore the HCIBox, however, I got the below error.

The below prerequisites were met
The az version is 2.45.0
A 100 ESv5-series vCPU 
All the necessary Azure providers added
The Azure service principle has the owner's permission at the subscription level
The main.parameters.json file is updated according to the service principle and tenant information

Could you please advise a possible solution?

{"status":"Failed","error":{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.","details":[{"code":"Conflict","message":"{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The 'AzureAsyncOperationWaiting' resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"DeploymentFailed\",\r\n        \"message\": \"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.\",\r\n        \"details\": [\r\n          {\r\n            \"code\": \"NotFound\",\r\n        
    \"message\": \"{\\r\\n  [\\\"error\\\](file:///%22error/)": {\\r\\n    [\\\"code\\\](file:///%22code/)": [\\\"BadRequest\\\",\\r\\n](file:///%22BadRequest/%22,/r/n)    [\\\"message\\\](file:///%22message/)": [\\\"\\\"\\r\\n](file:///%22/%22/r/n)  }\\r\\n}\"\r\n          }\r\n        ]\r\n      }\r\n    ]\r\n  }\r\n}"}]}}

Other information:
"correlationId": "c75f8af0-3887-4f6f-95b6-d6e336213e96",
    "description": "",
    "eventDataId": "064de4b9-bb77-40cc-99a4-2d37994013cd",
    "eventName": {
        "value": "EndRequest",
        "localizedValue": "End request"
    }

**Comment by dmc-tech** [comment]
If you try the deployment again and get the same error, it proves it is not a transient error and points toward a config issue
First things to check are: does the Resource group exist in the correct subscription? If so, when you check the deployments blade, does it show the attempted deployment(s) with the same error? (this at least shows you have the correct subscription set and Bicep can make the call to ARM)

**Comment by walezbee** [comment]
Dear dmc-tech
Thank you for getting back to me.

Yes, the resources group exists in the current subscription. 
I've attached the errors from the deployment blade.

[Deployment-main.zip](https://github.com/microsoft/azure_arc/files/10978527/Deployment-main.zip)
[Deployment-mgmtArtifactsAndPolicyDeployment.zip](https://github.com/microsoft/azure_arc/files/10978528/Deployment-mgmtArtifactsAndPolicyDeployment.zip)

**Reply by dmc-tech** [reply]
Sorry for the delay.
It looks like your logAnalyticsWorkspaceName has not been changed in the main.parameters.json file

```
          "message": "<!DOCTYPE html>\r\n<html>\r\n    <head>\r\n        <title>Runtime Error</title>\r\n        <meta name=\"viewport\" content=\"width=device-width\" />\r\n        <style>\r\n         body {font-family:\"Verdana\";font-weight:normal;font-size: .7em;color:black;} \r\n         p {font-family:\"Verdana\";font-weight:normal;color:black;margin-top: -5px}\r\n         b {font-family:\"Verdana\";font-weight:bold;color:black;margin-top: -5px}\r\n         H1 { font-family:\"Verdana\";font-weight:normal;font-size:18pt;color:red }\r\n         H2 { font-family:\"Verdana\";font-weight:normal;font-size:14pt;color:maroon }\r\n         pre {font-family:\"Consolas\",\"Lucida Console\",Monospace;font-size:11pt;margin:0;padding:0.5em;line-height:14pt}\r\n         .marker {font-weight: bold; color: black;text-decoration: none;}\r\n         .version {color: gray;}\r\n         .error {margin-bottom: 10px;}\r\n         .expandable { text-decoration:underline; font-weight:bold; color:navy; cursor:hand; }\r\n         @media screen and (max-width: 639px) {\r\n          pre { width: 440px; overflow: auto; white-space: pre-wrap; word-wrap: break-word; }\r\n         }\r\n         @media screen and (max-width: 479px) {\r\n          pre { width: 280px; }\r\n         }\r\n        </style>\r\n    </head>\r\n\r\n    <body bgcolor=\"white\">\r\n\r\n            <span><H1>Server Error in '/' Application.<hr width=100% size=1 color=silver></H1>\r\n\r\n            <h2> <i>Runtime Error</i> </h2></span>\r\n\r\n            <font face=\"Arial, Helvetica, Geneva, SunSans-Regular, sans-serif \">\r\n\r\n            <b> Description: </b>An application error occurred on the server. The current custom error settings for this application prevent the details of the application error from being viewed remotely (for security reasons). It could, however, be viewed by browsers running on the local server machine.\r\n            <br><br>\r\n\r\n            <b>Details:</b> To enable the details of this specific error message to be viewable on remote machines, please create a &lt;customErrors&gt; tag within a &quot;web.config&quot; configuration file located in the root directory of the current web application. This &lt;customErrors&gt; tag should then have its &quot;mode&quot; attribute set to &quot;Off&quot;.<br><br>\r\n\r\n            <table width=100% bgcolor=\"#ffffcc\">\r\n               <tr>\r\n                  <td>\r\n                      <code><pre>\r\n\r\n&lt;!-- Web.Config Configuration File --&gt;\r\n\r\n&lt;configuration&gt;\r\n    &lt;system.web&gt;\r\n        &lt;customErrors mode=&quot;Off&quot;/&gt;\r\n    &lt;/system.web&gt;\r\n&lt;/configuration&gt;</pre></code>\r\n\r\n                  </td>\r\n               </tr>\r\n            </table>\r\n\r\n            <br>\r\n\r\n            <b>Notes:</b> The current error page you are seeing can be replaced by a custom error page by modifying the &quot;defaultRedirect&quot; attribute of the application&#39;s &lt;customErrors&gt; configuration tag to point to a custom error page URL.<br><br>\r\n\r\n            <table width=100% bgcolor=\"#ffffcc\">\r\n               <tr>\r\n                  <td>\r\n                      <code><pre>\r\n\r\n&lt;!-- Web.Config Configuration File --&gt;\r\n\r\n&lt;configuration&gt;\r\n    &lt;system.web&gt;\r\n        &lt;customErrors mode=&quot;RemoteOnly&quot; defaultRedirect=&quot;mycustompage.htm&quot;/&gt;\r\n    &lt;/system.web&gt;\r\n&lt;/configuration&gt;</pre></code>\r\n\r\n                  </td>\r\n               </tr>\r\n            </table>\r\n\r\n            <br>\r\n\r\n    </body>\r\n</html>\r\n"
        }
      },
      "targetResource": {
        "id": "/subscriptions/59452154-348f-46b7-b159-aec055b5784e/resourceGroups/HCIBox-Rg/providers/Microsoft.OperationalInsights/workspaces/<your unique Log Analytics workspace name>",
        "resourceType": "Microsoft.OperationalInsights/workspaces",
        "resourceName": "<your unique Log Analytics workspace name>"
      }
    }
````

Make sure it doesn't have the <> chars in the name and it should run through.

**Comment by walezbee** [comment, marked as answer]
Dear Danny,
Thank you for your email.

I actually changed the logAnalyticsWorkspaceName. However, I changed the
name from *ExampleWs* to *Examplews*, which was successfully deployed.

Thank you for being so supportive.
